File: android-java/BottomNavAPP/app/src/main/java/com/tnqr/bottomnavapp/MainActivity.java
package com.tnqr.bottomnavapp;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.tnqr.bottomnavapp.databinding.ActivityMainBinding;
import com.tnqr.bottomnavapp.fragments.AccountFragment;
import com.tnqr.bottomnavapp.fragments.HomeFragment;
import com.tnqr.bottomnavapp.fragments.Settings_Fragment;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view =  binding.getRoot();
        setContentView(view);


        binding.bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                int id = item.getItemId();

                if(id == R.id.home)
                    loadFragment(new HomeFragment());
                else if (id==R.id.account)
                    loadFragment(new AccountFragment());
                else if (id==R.id.settings)
                    loadFragment(new Settings_Fragment());

                return false;
            }
        });
    }
    private void loadFragment(Fragment fragment){
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.frame_layout,fragment);
        fragmentTransaction.commit();
    }
}

File: android-kotlin/RetrofitTest/app/src/main/java/com/example/retrofittest/MainActivity.kt
package com.example.retrofittest

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.retrofittest.adapter.CityAdapter
import com.example.retrofittest.databinding.ActivityMainBinding
import com.example.retrofittest.model.CityData

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    private lateinit var cityAdapter : CityAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view  = binding.root
        setContentView(view)

        val cityList = intent.getParcelableArrayListExtra<CityData>("city_list") ?: arrayListOf()

        // Eğer veri gelmişse RecyclerView'e bağlayacağız
        cityAdapter = CityAdapter(cityList)
        binding.recyclerViewCities.adapter = cityAdapter


    }
}